src/binary-reader-ir.cc

```diff
@@ -1125,6 +1143,7 @@ Result BinaryReaderIR::OnRefFuncExpr(Index func_index) {
 }
 
 Result BinaryReaderIR::OnRefNullExpr(Type type) {
+  module_->features_used.exceptions |= (type == Type::ExnRef);
```

Comment:
Is this needed?  Wouldn't the Type have to be declared in the type section in order to produce this?   (i.e. isn't' this already handled the by the type section parsing?

Reply:
if we're not wrong, our understanding is that `ref.null` calls for a `reftype`, not a type section `typeidx`.

e.g. this is valid

```
(module
  (func
    ref.null exn
    drop
  )
)
```

so anything that produces `exnref` must set `features_used.exceptions`, that way `wasm2c` knows to include their definition. (meanwhile something like `throw_ref` doesn't need to set it.)

Reply:
Sounds reasonable.